### PR TITLE
feat: support al 2023

### PR DESCRIPTION
**Pull request by denise89p**

### CODE REVIEW

File: lambdas/aws_functions_def/functions_def.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (d *AWSFuncDef) GetFunctionCmdDefinition(name functions_def.FunctionName) s
 		funcDefTemplate := `
 		function %s {
 			local json_data=$1
-			res=$(aws lambda invoke --region %s --function-name %s --payload "$json_data" output)
+			res=$(aws lambda invoke --region %s --function-name %s --cli-binary-format raw-in-base64-out --payload "$json_data" output)
```
Comment (assafgi) [inline lambdas/aws_functions_def/functions_def.go:56]: will it break amazon linux 2? we should still support it

File: modules/clients/init.sh [hunk 2]
```diff
@@ -56,6 +56,7 @@ function setup_aws_logs_agent() {
 setup_aws_logs_agent || echo "Failed to setup AWS logs agent"
 
 yum -y install pip || true
+yum -y install kernel-devel || true
```
Comment (assafgi) [inline modules/clients/init.sh:59]: why do we need it?

File: modules/clients/mount_wekafs.sh [hunk 3]
```diff
@@ -68,15 +68,15 @@ weka local stop && weka local rm -f --all
 
 FRONTEND_CONTAINER_CORES_NUM="${frontend_container_cores_num}"
 NICS_NUM=$((FRONTEND_CONTAINER_CORES_NUM+1))
-first_interface=$(ip -o link show | awk -F ': ' '!/docker0/ && !/^lo/ {print $2}' | sort | head -n 1)
+first_interface=$(ip -o link show | awk -F ': ' '!/docker0/ && !/lo/ {print $2}' | head -n 1)
 interface_str=$(echo $first_interface | awk '{gsub(/[0-9]/,"",$1); print $1}')
```
Comment (assafgi) [inline modules/clients/mount_wekafs.sh:72]: See my comment on go-cloud-lib about simpler bash manipulation

File: modules/clients/init.sh (hunk 2 above)
Comment (denise89p) [reply modules/clients/init.sh:59]: `^@error: Driver command failed with error code 1
Error: Unable to find a match: kernel-devel-6.1.106-116.188.amzn2023.x86_64
Amazon Linux 2023 repository                    1.3 kB/s | 243  B     00:00    
Errors during downloading metadata for repository 'amazonlinux':
  - Status code: 403 for https://al2023-repos-eu-west-1-de612dc2.s3.dualstack.eu-west-1.amazonaws.com/core/mirrors/2023/x86_64/mirror.list (IP: 52.92.32.186)
  - Status code: 403 for https://al2023-repos-eu-west-1-de612dc2.s3.dualstack.eu-west-1.amazonaws.com/core/mirrors/2023/x86_64/mirror.list (IP: 3.5.65.151)
Error: Failed to download metadata for repo 'amazonlinux': Cannot prepare internal mirrorlist: Status code: 403 for https://al2023-repos-eu-west-1-de612dc2.s3.dualstack.eu-west-1.amazonaws.com/core/mirrors/2023/x86_64/mirror.list (IP: 52.92.32.186)
Ignoring repositories: amazonlinux
Last metadata expiration check: 0:01:23 ago on Sun Sep  8 07:35:05 2024.
No match for argument: kernel-devel-6.1.106-116.188.amzn2023.x86_64
Error: Unable to find a match: kernel-devel-6.1.106-116.188.amzn2023.x86_64
:
`

File: modules/clients/mount_wekafs.sh [hunk 4]
```diff
@@ -68,18 +68,18 @@ weka local stop && weka local rm -f --all
...
 
 function getNetStrForDpdk() {
 	i=$1
 	j=$2
-
+	interface_name=$3
 	net=""
 	for ((i; i<$j; i++)); do
-		net="$net -o net=$interface_str$i"
+		net="$net -o net=$interface_name$i"
```
Comment (assafgi) [inline modules/clients/mount_wekafs.sh:82]: should be `interface_base_name`

**Review by assafgi — APPROVED** [review]

**Pull request merged**